dev/fattest.simplicity/src/componenttest/rules/repeater/JakartaEE9Action.java

```diff
@@ -177,8 +190,11 @@ public void setup() throws Exception {
         }
         ShrinkHelper.cleanAllExportedArchives();
 
-        // Transform server.xml's
         super.setup();
+
+        if (transformServerConfig)
+            for (File serverXml : super.getReplacementServers())
```

Comment:
This may be a bit inefficient as we need to call the transformer multiple times in cases when there are multiple server.xml's.  I wonder if we can provide a folder with all the server.xml's instead.

Reply:
Adding a folder would unnecessarily impart structure on the FAT project or require gradle changes.  I'm finding that few of the server and client configurations (let's just say `server.xml` files) in a FAT suite need transformation of config elements other than `<featureManager>`.  Invoking the transformer is expensive.  Rather than compute the set `serverXmls` I think the user should specify this set as an argument of method `transformServerConfig()` [or not -- see (*) below.]  The repeat action would also need to know which transform properties to apply to each file in the set.  The user would have to input this information as well, which IMHO adds unnecessary complication.  I think it's better that the user adds the server.xml files to the `jakartaee-dd-master` properties file, and whenever a server.xml file is specified, the repeat action simply invokes existing method `transformApp()` on the file.  This approach ensures the transformer is invoked uniformly across all product and test artifacts -- e.g. product configuration type metatype.xml and any corresponding config elements in server.xml -- as expected per each jakarta technology.  The approach alleviates the user from adding code to transform a server.xml while otherwise maintaining the usage pattern of how all artifacts are transformed. 

(*) There's another benefit to requiring the user to add server.xml files to the jakartaee-dd-master properties file: the need for method `transformServerConfig()` is obviated.  The repeat action can automagically invoke transformApp on any server.xml file it finds in the jakartaee-dd-master properties file. -- i.e. the serverXml set is implicitly configured by the user.

Reply:
The problem is that the `transformApp()` method is called when an application is shrink-wrapped and deployed to server at runtime. The server.xml file is not a part of that process so adding those server.xml files to `jakartaee-dd-master` would never be found/transformed.

So we still need a different method of collecting a list of server.xml files that need to be transformed, there isn't a way (that I know of) to include this in the current app transformation process. 

Additionally users would still need to ensure the transformations they want are somewhere in the `wlp-jakartaee-transform/rules` folder.